openmsupply/open-msupply
# Outbound shipment auto allocation

Fixes #2227, #2266, #2264, #1983, #2255

# 👩🏻‍💻 What does this PR do? 
 
See #2261 for the discussion and requested changes. Reinstates the auto allocation which was removed in #1189

- Allow saving of lines with 0 quantity
- Show a warning if you click OK when the quantity is 0; this asks you to click OK again to confirm
- Reinstate the auto allocation, this happens when the issue quantity is changed or the pack size is changed
- Prune 0 quantity lines when changing status from NEW
- Show a warning when changing status
- Fix the strange warning issue - where the notification shows incorrectly
- Show the warning when the pack size prevents the requested quantity being issued
- Show a warning when a placeholder has been allocated due to insufficient stock


<img width="1060" alt="Screenshot 2023-09-27 at 11 39 12 AM" src="https://github.com/openmsupply/open-msupply/assets/9192912/d7b8c100-8776-425f-b722-16ac239af774">

<img width="1056" alt="Screenshot 2023-09-27 at 11 38 44 AM" src="https://github.com/openmsupply/open-msupply/assets/9192912/c2bf8813-36dd-43b3-89d7-2ce23b818b9f">

![Screenshot 2023-09-27 at 10 31 44 PM](https://github.com/openmsupply/open-msupply/assets/9192912/023e1337-feda-45d4-a09d-a235ace30355)

![Screenshot 2023-09-27 at 7 10 23 PM](https://github.com/openmsupply/open-msupply/assets/9192912/7a3f870b-64ed-4de4-be78-0a2577531716)




# 🧪 How has/should this change been tested? 


- Create a new outbound shipment and try to add an item with 0 quantity
- Update an existing line to 0  and check the messaging
- Try the auto allocation
- Try the two way updates - editing a line and checking that the issue quantity updates
- Change status and check that lines are pruned


## 💌 Any notes for the reviewer?


## 📃 Documentation

- [ ]  Auto allocation
